File: clients/ios/Lextures/Core/LMS/CourseFeaturesLogic.swift
import Foundation

/// Course features / tools helpers (M13.2).
enum CourseFeaturesLogic {
    enum Tool: String, CaseIterable, Identifiable, Hashable {
        case adaptivePaths
        case aiTutor
        case attendance
        case calendar
        case collabDocs
        case sections
        case discussions
        case feed
        case files
        case liveSessions
        case misconceptionDetection
        case multilingualMessaging
        case notebook
        case officeHours
        case diagnosticAssessments
        case questionBank
        case reportCards
        case hintScaffolding
        case lockdownMode
        case srs
        case standardsAlignment
        case visualBoards
        case whiteboard

        var id: String { rawValue }

        var labelKey: String { "mobile.courseSettings.features.tool.\(rawValue).label" }
        var descriptionKey: String { "mobile.courseSettings.features.tool.\(rawValue).description" }
    }

    struct ToolRow: Identifiable, Hashable {
        var tool: Tool
        var id: String { tool.id }
    }

    static let allToolRows: [ToolRow] = [
        .init(tool: .adaptivePaths),
        .init(tool: .aiTutor),
        .init(tool: .attendance),
        .init(tool: .calendar),
        .init(tool: .collabDocs),
        .init(tool: .sections),
        .init(tool: .discussions),
        .init(tool: .feed),
        .init(tool: .files),
        .init(tool: .liveSessions),
        .init(tool: .misconceptionDetection),
        .init(tool: .multilingualMessaging),
        .init(tool: .notebook),
        .init(tool: .officeHours),
        .init(tool: .diagnosticAssessments),
        .init(tool: .questionBank),
        .init(tool: .reportCards),
        .init(tool: .hintScaffolding),
        .init(tool: .lockdownMode),
        .init(tool: .srs),
        .init(tool: .standardsAlignment),
        .init(tool: .visualBoards),
        .init(tool: .whiteboard),
    ]

    static func filterTools(_ tools: [ToolRow], query: String) -> [ToolRow] {
        let trimmed = query.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return tools }
        let queryLower = trimmed.lowercased()
        return tools.filter { row in
            L.text(String.LocalizationValue(row.tool.labelKey)).lowercased().contains(queryLower)
                || L.text(String.LocalizationValue(row.tool.descriptionKey)).lowercased().contains(queryLower)
        }
    }

    private static let enabledReaders: [Tool: (CourseSummary) -> Bool] = [
        .adaptivePaths: { $0.adaptivePathsEnabled == true },
        .aiTutor: { $0.aiTutorEnabled == true },
        .attendance: { $0.attendanceEnabled == true },
        .calendar: { $0.calendarEnabled != false },
        .collabDocs: { $0.collabDocsEnabled == true },
        .sections: { $0.sectionsEnabled == true },
        .discussions: { $0.discussionsEnabled == true },
        .feed: { $0.feedEnabled != false },
        .files: { $0.filesEnabled != false },
        .liveSessions: { $0.liveSessionsEnabled == true },
        .misconceptionDetection: { $0.misconceptionDetectionEnabled == true },
        .multilingualMessaging: { $0.multilingualMessagingEnabled == true },
        .notebook: { $0.notebookEnabled != false },
        .officeHours: { $0.officeHoursEnabled == true },
        .diagnosticAssessments: { $0.diagnosticAssessmentsEnabled == true },
        .questionBank: { $0.questionBankEnabled == true },
        .reportCards: { $0.reportCardsEnabled == true },
        .hintScaffolding: { $0.hintScaffoldingEnabled == true },
        .lockdownMode: { $0.lockdownModeEnabled == true },
        .srs: { $0.srsEnabled == true },
        .standardsAlignment: { $0.standardsAlignmentEnabled == true },
        .visualBoards: { $0.visualBoardsEnabled == true },
        .whiteboard: { $0.whiteboardEnabled == true },
    ]

    private static let togglers: [Tool: (inout CourseSummary, Bool) -> Void] = [
        .adaptivePaths: { $0.adaptivePathsEnabled = $1 },
        .aiTutor: { $0.aiTutorEnabled = $1 },
        .attendance: { $0.attendanceEnabled = $1 },
        .calendar: { $0.calendarEnabled = $1 },
        .collabDocs: { $0.collabDocsEnabled = $1 },
        .sections: { $0.sectionsEnabled = $1 },
        .discussions: { $0.discussionsEnabled = $1 },
        .feed: { $0.feedEnabled = $1 },
        .files: { $0.filesEnabled = $1 },
        .liveSessions: { $0.liveSessionsEnabled = $1 },
        .misconceptionDetection: { $0.misconceptionDetectionEnabled = $1 },
        .multilingualMessaging: { $0.multilingualMessagingEnabled = $1 },
        .notebook: { $0.notebookEnabled = $1 },
        .officeHours: { $0.officeHoursEnabled = $1 },
        .diagnosticAssessments: { $0.diagnosticAssessmentsEnabled = $1 },
        .questionBank: { $0.questionBankEnabled = $1 },
        .reportCards: { $0.reportCardsEnabled = $1 },
        .hintScaffolding: { $0.hintScaffoldingEnabled = $1 },
        .lockdownMode: { $0.lockdownModeEnabled = $1 },
        .srs: { $0.srsEnabled = $1 },
        .standardsAlignment: { $0.standardsAlignmentEnabled = $1 },
        .visualBoards: { $0.visualBoardsEnabled = $1 },
        .whiteboard: { $0.whiteboardEnabled = $1 },
    ]

    static func isEnabled(_ tool: Tool, course: CourseSummary) -> Bool {
        enabledReaders[tool]?(course) ?? false
    }

    static func applyToggle(course: CourseSummary, tool: Tool, enabled: Bool) -> CourseSummary {
        var updated = course
        togglers[tool]?(&updated, enabled)
        return updated
    }

    static func buildFeaturesPatch(from course: CourseSummary) -> CourseFeaturesPatch {
        CourseFeaturesPatch(
            notebookEnabled: course.notebookEnabled != false,
            feedEnabled: course.feedEnabled != false,
            calendarEnabled: course.calendarEnabled != false,
            questionBankEnabled: course.questionBankEnabled == true,
            lockdownModeEnabled: course.lockdownModeEnabled == true,
            standardsAlignmentEnabled: course.standardsAlignmentEnabled == true,
            adaptivePathsEnabled: course.adaptivePathsEnabled == true,
            srsEnabled: course.srsEnabled == true,
            diagnosticAssessmentsEnabled: course.diagnosticAssessmentsEnabled == true,
            hintScaffoldingEnabled: course.hintScaffoldingEnabled == true,
            misconceptionDetectionEnabled: course.misconceptionDetectionEnabled == true,
            sectionsEnabled: course.sectionsEnabled == true,
            discussionsEnabled: course.discussionsEnabled == true,
            collabDocsEnabled: course.collabDocsEnabled == true,
            liveSessionsEnabled: course.liveSessionsEnabled == true,
            officeHoursEnabled: course.officeHoursEnabled == true,
            aiTutorEnabled: course.aiTutorEnabled == true,
            multilingualMessagingEnabled: course.multilingualMessagingEnabled == true,
            filesEnabled: course.filesEnabled != false,
            attendanceEnabled: course.attendanceEnabled == true,
            whiteboardEnabled: course.whiteboardEnabled == true,
            reportCardsEnabled: course.reportCardsEnabled == true,
            visualBoardsEnabled: course.visualBoardsEnabled == true
        )
    }

    static func shouldConfirmDisable(_ tool: Tool, currentlyEnabled: Bool) -> Bool {
        currentlyEnabled
    }

    static func videoCaptionsSectionEnabled(_ features: MobilePlatformFeatures) -> Bool {
        features.videoCaptionsEnabled
    }

    static func consortiumSectionEnabled(_ features: MobilePlatformFeatures) -> Bool {
        features.ffConsortiumSharing
    }

    static func cacheKeyFeatures(courseCode: String) -> String {
        "course:\(courseCode):features"
    }

    static func cacheKeyConsortium(courseCode: String) -> String {
        "course:\(courseCode):consortium"
    }

    static func toggleIdempotencyKey(courseCode: String, tool: Tool) -> String {
        "course-features:\(courseCode):\(tool.rawValue)"
    }

    static func captionPolicyIdempotencyKey(courseCode: String) -> String {
        "course-caption-policy:\(courseCode)"
    }

    static func consortiumIdempotencyKey(courseCode: String) -> String {
        "course-consortium:\(courseCode)"
    }

    static func userFacingError(_ error: Error) -> String {
        if let apiError = error as? APIError {
            return apiError.errorDescription ?? L.text("mobile.courseSettings.features.genericError")
        }
        return error.localizedDescription
    }
}

struct CourseFeaturesPatch: Codable, Equatable {
    var notebookEnabled: Bool
    var feedEnabled: Bool
    var calendarEnabled: Bool
    var questionBankEnabled: Bool
    var lockdownModeEnabled: Bool
    var standardsAlignmentEnabled: Bool
    var adaptivePathsEnabled: Bool
    var srsEnabled: Bool
    var diagnosticAssessmentsEnabled: Bool
    var hintScaffoldingEnabled: Bool
    var misconceptionDetectionEnabled: Bool
    var sectionsEnabled: Bool
    var discussionsEnabled: Bool
    var collabDocsEnabled: Bool
    var liveSessionsEnabled: Bool
    var officeHoursEnabled: Bool
    var aiTutorEnabled: Bool
    var multilingualMessagingEnabled: Bool
    var filesEnabled: Bool
    var attendanceEnabled: Bool
    var whiteboardEnabled: Bool
    var reportCardsEnabled: Bool
    var visualBoardsEnabled: Bool
}

struct CourseCaptionPolicyPatch: Codable, Equatable {
    var requireCaptions: Bool
}

struct CourseConsortiumSettings: Codable, Equatable {
    var consortiumShareable: Bool
}

struct CourseConsortiumSettingsPatch: Codable, Equatable {
    var consortiumShareable: Bool
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/CourseFeaturesLogic.kt
package com.lextures.android.core.lms

import com.lextures.android.core.navigation.MobilePlatformFeatures

/** Course features / tools helpers (M13.2). */
object CourseFeaturesLogic {
    enum class Tool(val id: String) {
        adaptivePaths("adaptivePaths"),
        aiTutor("aiTutor"),
        attendance("attendance"),
        calendar("calendar"),
        collabDocs("collabDocs"),
        sections("sections"),
        discussions("discussions"),
        feed("feed"),
        files("files"),
        liveSessions("liveSessions"),
        misconceptionDetection("misconceptionDetection"),
        multilingualMessaging("multilingualMessaging"),
        notebook("notebook"),
        officeHours("officeHours"),
        diagnosticAssessments("diagnosticAssessments"),
        questionBank("questionBank"),
        reportCards("reportCards"),
        hintScaffolding("hintScaffolding"),
        lockdownMode("lockdownMode"),
        srs("srs"),
        standardsAlignment("standardsAlignment"),
        visualBoards("visualBoards"),
        whiteboard("whiteboard"),
    }

    data class ToolRow(val tool: Tool)

    val allToolRows: List<ToolRow> = listOf(
        ToolRow(Tool.adaptivePaths),
        ToolRow(Tool.aiTutor),
        ToolRow(Tool.attendance),
        ToolRow(Tool.calendar),
        ToolRow(Tool.collabDocs),
        ToolRow(Tool.sections),
        ToolRow(Tool.discussions),
        ToolRow(Tool.feed),
        ToolRow(Tool.files),
        ToolRow(Tool.liveSessions),
        ToolRow(Tool.misconceptionDetection),
        ToolRow(Tool.multilingualMessaging),
        ToolRow(Tool.notebook),
        ToolRow(Tool.officeHours),
        ToolRow(Tool.diagnosticAssessments),
        ToolRow(Tool.questionBank),
        ToolRow(Tool.reportCards),
        ToolRow(Tool.hintScaffolding),
        ToolRow(Tool.lockdownMode),
        ToolRow(Tool.srs),
        ToolRow(Tool.standardsAlignment),
        ToolRow(Tool.visualBoards),
        ToolRow(Tool.whiteboard),
    )

    fun filterTools(tools: List<ToolRow>, query: String): List<ToolRow> {
        val trimmed = query.trim()
        if (trimmed.isEmpty()) return tools
        val q = trimmed.lowercase()
        return tools.filter { row ->
            toolLabelKey(row.tool).lowercase().contains(q) ||
                toolDescriptionKey(row.tool).lowercase().contains(q)
        }
    }

    fun toolLabelRes(tool: Tool): Int = when (tool) {
        Tool.adaptivePaths -> com.lextures.android.R.string.mobile_courseSettings_features_tool_adaptivePaths_label
        Tool.aiTutor -> com.lextures.android.R.string.mobile_courseSettings_features_tool_aiTutor_label
        Tool.attendance -> com.lextures.android.R.string.mobile_courseSettings_features_tool_attendance_label
        Tool.calendar -> com.lextures.android.R.string.mobile_courseSettings_features_tool_calendar_label
        Tool.collabDocs -> com.lextures.android.R.string.mobile_courseSettings_features_tool_collabDocs_label
        Tool.sections -> com.lextures.android.R.string.mobile_courseSettings_features_tool_sections_label
        Tool.discussions -> com.lextures.android.R.string.mobile_courseSettings_features_tool_discussions_label
        Tool.feed -> com.lextures.android.R.string.mobile_courseSettings_features_tool_feed_label
        Tool.files -> com.lextures.android.R.string.mobile_courseSettings_features_tool_files_label
        Tool.liveSessions -> com.lextures.android.R.string.mobile_courseSettings_features_tool_liveSessions_label
        Tool.misconceptionDetection -> com.lextures.android.R.string.mobile_courseSettings_features_tool_misconceptionDetection_label
        Tool.multilingualMessaging -> com.lextures.android.R.string.mobile_courseSettings_features_tool_multilingualMessaging_label
        Tool.notebook -> com.lextures.android.R.string.mobile_courseSettings_features_tool_notebook_label
        Tool.officeHours -> com.lextures.android.R.string.mobile_courseSettings_features_tool_officeHours_label
        Tool.diagnosticAssessments -> com.lextures.android.R.string.mobile_courseSettings_features_tool_diagnosticAssessments_label
        Tool.questionBank -> com.lextures.android.R.string.mobile_courseSettings_features_tool_questionBank_label
        Tool.reportCards -> com.lextures.android.R.string.mobile_courseSettings_features_tool_reportCards_label
        Tool.hintScaffolding -> com.lextures.android.R.string.mobile_courseSettings_features_tool_hintScaffolding_label
        Tool.lockdownMode -> com.lextures.android.R.string.mobile_courseSettings_features_tool_lockdownMode_label
        Tool.srs -> com.lextures.android.R.string.mobile_courseSettings_features_tool_srs_label
        Tool.standardsAlignment -> com.lextures.android.R.string.mobile_courseSettings_features_tool_standardsAlignment_label
        Tool.visualBoards -> com.lextures.android.R.string.mobile_courseSettings_features_tool_visualBoards_label
        Tool.whiteboard -> com.lextures.android.R.string.mobile_courseSettings_features_tool_whiteboard_label
    }

    fun toolDescriptionRes(tool: Tool): Int = when (tool) {
        Tool.adaptivePaths -> com.lextures.android.R.string.mobile_courseSettings_features_tool_adaptivePaths_description
        Tool.aiTutor -> com.lextures.android.R.string.mobile_courseSettings_features_tool_aiTutor_description
        Tool.attendance -> com.lextures.android.R.string.mobile_courseSettings_features_tool_attendance_description
        Tool.calendar -> com.lextures.android.R.string.mobile_courseSettings_features_tool_calendar_description
        Tool.collabDocs -> com.lextures.android.R.string.mobile_courseSettings_features_tool_collabDocs_description
        Tool.sections -> com.lextures.android.R.string.mobile_courseSettings_features_tool_sections_description
        Tool.discussions -> com.lextures.android.R.string.mobile_courseSettings_features_tool_discussions_description
        Tool.feed -> com.lextures.android.R.string.mobile_courseSettings_features_tool_feed_description
        Tool.files -> com.lextures.android.R.string.mobile_courseSettings_features_tool_files_description
        Tool.liveSessions -> com.lextures.android.R.string.mobile_courseSettings_features_tool_liveSessions_description
        Tool.misconceptionDetection -> com.lextures.android.R.string.mobile_courseSettings_features_tool_misconceptionDetection_description
        Tool.multilingualMessaging -> com.lextures.android.R.string.mobile_courseSettings_features_tool_multilingualMessaging_description
        Tool.notebook -> com.lextures.android.R.string.mobile_courseSettings_features_tool_notebook_description
        Tool.officeHours -> com.lextures.android.R.string.mobile_courseSettings_features_tool_officeHours_description
        Tool.diagnosticAssessments -> com.lextures.android.R.string.mobile_courseSettings_features_tool_diagnosticAssessments_description
        Tool.questionBank -> com.lextures.android.R.string.mobile_courseSettings_features_tool_questionBank_description
        Tool.reportCards -> com.lextures.android.R.string.mobile_courseSettings_features_tool_reportCards_description
        Tool.hintScaffolding -> com.lextures.android.R.string.mobile_courseSettings_features_tool_hintScaffolding_description
        Tool.lockdownMode -> com.lextures.android.R.string.mobile_courseSettings_features_tool_lockdownMode_description
        Tool.srs -> com.lextures.android.R.string.mobile_courseSettings_features_tool_srs_description
        Tool.standardsAlignment -> com.lextures.android.R.string.mobile_courseSettings_features_tool_standardsAlignment_description
        Tool.visualBoards -> com.lextures.android.R.string.mobile_courseSettings_features_tool_visualBoards_description
        Tool.whiteboard -> com.lextures.android.R.string.mobile_courseSettings_features_tool_whiteboard_description
    }

    private fun toolLabelKey(tool: Tool): String = tool.id

    private fun toolDescriptionKey(tool: Tool): String = "${tool.id}.description"

    fun isEnabled(tool: Tool, course: CourseSummary): Boolean = when (tool) {
        Tool.adaptivePaths -> course.adaptivePathsEnabled == true
        Tool.aiTutor -> course.aiTutorEnabled == true
        Tool.attendance -> course.attendanceEnabled == true
        Tool.calendar -> course.calendarEnabled != false
        Tool.collabDocs -> course.collabDocsEnabled == true
        Tool.sections -> course.sectionsEnabled == true
        Tool.discussions -> course.discussionsEnabled == true
        Tool.feed -> course.feedEnabled != false
        Tool.files -> course.filesEnabled != false
        Tool.liveSessions -> course.liveSessionsEnabled == true
        Tool.misconceptionDetection -> course.misconceptionDetectionEnabled == true
        Tool.multilingualMessaging -> course.multilingualMessagingEnabled == true
        Tool.notebook -> course.notebookEnabled != false
        Tool.officeHours -> course.officeHoursEnabled == true
        Tool.diagnosticAssessments -> course.diagnosticAssessmentsEnabled == true
        Tool.questionBank -> course.questionBankEnabled == true
        Tool.reportCards -> course.reportCardsEnabled == true
        Tool.hintScaffolding -> course.hintScaffoldingEnabled == true
        Tool.lockdownMode -> course.lockdownModeEnabled == true
        Tool.srs -> course.srsEnabled == true
        Tool.standardsAlignment -> course.standardsAlignmentEnabled == true
        Tool.visualBoards -> course.visualBoardsEnabled == true
        Tool.whiteboard -> course.whiteboardEnabled == true
    }

    fun applyToggle(course: CourseSummary, tool: Tool, enabled: Boolean): CourseSummary = when (tool) {
        Tool.adaptivePaths -> course.copy(adaptivePathsEnabled = enabled)
        Tool.aiTutor -> course.copy(aiTutorEnabled = enabled)
        Tool.attendance -> course.copy(attendanceEnabled = enabled)
        Tool.calendar -> course.copy(calendarEnabled = enabled)
        Tool.collabDocs -> course.copy(collabDocsEnabled = enabled)
        Tool.sections -> course.copy(sectionsEnabled = enabled)
        Tool.discussions -> course.copy(discussionsEnabled = enabled)
        Tool.feed -> course.copy(feedEnabled = enabled)
        Tool.files -> course.copy(filesEnabled = enabled)
        Tool.liveSessions -> course.copy(liveSessionsEnabled = enabled)
        Tool.misconceptionDetection -> course.copy(misconceptionDetectionEnabled = enabled)
        Tool.multilingualMessaging -> course.copy(multilingualMessagingEnabled = enabled)
        Tool.notebook -> course.copy(notebookEnabled = enabled)
        Tool.officeHours -> course.copy(officeHoursEnabled = enabled)
        Tool.diagnosticAssessments -> course.copy(diagnosticAssessmentsEnabled = enabled)
        Tool.questionBank -> course.copy(questionBankEnabled = enabled)
        Tool.reportCards -> course.copy(reportCardsEnabled = enabled)
        Tool.hintScaffolding -> course.copy(hintScaffoldingEnabled = enabled)
        Tool.lockdownMode -> course.copy(lockdownModeEnabled = enabled)
        Tool.srs -> course.copy(srsEnabled = enabled)
        Tool.standardsAlignment -> course.copy(standardsAlignmentEnabled = enabled)
        Tool.visualBoards -> course.copy(visualBoardsEnabled = enabled)
        Tool.whiteboard -> course.copy(whiteboardEnabled = enabled)
    }

    fun buildFeaturesPatch(course: CourseSummary): CourseFeaturesPatch = CourseFeaturesPatch(
        notebookEnabled = course.notebookEnabled != false,
        feedEnabled = course.feedEnabled != false,
        calendarEnabled = course.calendarEnabled != false,
        questionBankEnabled = course.questionBankEnabled == true,
        lockdownModeEnabled = course.lockdownModeEnabled == true,
        standardsAlignmentEnabled = course.standardsAlignmentEnabled == true,
        adaptivePathsEnabled = course.adaptivePathsEnabled == true,
        srsEnabled = course.srsEnabled == true,
        diagnosticAssessmentsEnabled = course.diagnosticAssessmentsEnabled == true,
        hintScaffoldingEnabled = course.hintScaffoldingEnabled == true,
        misconceptionDetectionEnabled = course.misconceptionDetectionEnabled == true,
        sectionsEnabled = course.sectionsEnabled == true,
        discussionsEnabled = course.discussionsEnabled == true,
        collabDocsEnabled = course.collabDocsEnabled == true,
        liveSessionsEnabled = course.liveSessionsEnabled == true,
        officeHoursEnabled = course.officeHoursEnabled == true,
        aiTutorEnabled = course.aiTutorEnabled == true,
        multilingualMessagingEnabled = course.multilingualMessagingEnabled == true,
        filesEnabled = course.filesEnabled != false,
        attendanceEnabled = course.attendanceEnabled == true,
        whiteboardEnabled = course.whiteboardEnabled == true,
        reportCardsEnabled = course.reportCardsEnabled == true,
        visualBoardsEnabled = course.visualBoardsEnabled == true,
    )

    fun shouldConfirmDisable(currentlyEnabled: Boolean): Boolean = currentlyEnabled

    fun videoCaptionsSectionEnabled(features: MobilePlatformFeatures): Boolean = features.videoCaptionsEnabled

    fun consortiumSectionEnabled(features: MobilePlatformFeatures): Boolean = features.ffConsortiumSharing

    fun cacheKeyFeatures(courseCode: String): String = "course:$courseCode:features"

    fun cacheKeyConsortium(courseCode: String): String = "course:$courseCode:consortium"

    fun toggleIdempotencyKey(courseCode: String, tool: Tool): String = "course-features:$courseCode:${tool.id}"

    fun captionPolicyIdempotencyKey(courseCode: String): String = "course-caption-policy:$courseCode"

    fun consortiumIdempotencyKey(courseCode: String): String = "course-consortium:$courseCode"
}

@kotlinx.serialization.Serializable
data class CourseFeaturesPatch(
    val notebookEnabled: Boolean,
    val feedEnabled: Boolean,
    val calendarEnabled: Boolean,
    val questionBankEnabled: Boolean,
    val lockdownModeEnabled: Boolean,
    val standardsAlignmentEnabled: Boolean,
    val adaptivePathsEnabled: Boolean,
    val srsEnabled: Boolean,
    val diagnosticAssessmentsEnabled: Boolean,
    val hintScaffoldingEnabled: Boolean,
    val misconceptionDetectionEnabled: Boolean,
    val sectionsEnabled: Boolean,
    val discussionsEnabled: Boolean,
    val collabDocsEnabled: Boolean,
    val liveSessionsEnabled: Boolean,
    val officeHoursEnabled: Boolean,
    val aiTutorEnabled: Boolean,
    val multilingualMessagingEnabled: Boolean,
    val filesEnabled: Boolean,
    val attendanceEnabled: Boolean,
    val whiteboardEnabled: Boolean,
    val reportCardsEnabled: Boolean,
    val visualBoardsEnabled: Boolean,
)

@kotlinx.serialization.Serializable
data class CourseCaptionPolicyPatch(
    val requireCaptions: Boolean,
)

@kotlinx.serialization.Serializable
data class CourseConsortiumSettings(
    val consortiumShareable: Boolean = false,
)

@kotlinx.serialization.Serializable
data class CourseConsortiumSettingsPatch(
    val consortiumShareable: Boolean,
)
